prime-router/src/main/kotlin/fhirengine/azure/FHIRFlowFunctions.kt

```diff
@@ -54,8 +61,20 @@ class FHIRFlowFunctions : Logging {
             responseBuilder.body(body)
         } catch (e: IllegalArgumentException) {
             responseBuilder.body(e.message)
+            actionHistory.trackLogs(
+                ActionLog(InvalidTranslationMessage(e.message ?: "Translation failure."), type = ActionLogLevel.error)
+            )
         }
-        return responseBuilder.build()
+
+        val workflowEngine = WorkflowEngine.Builder().metadata(Metadata.getInstance()).build()
+        val response = responseBuilder.build()
+
+        // despite similar names, these fns save different data
+        actionHistory.trackActionRequestResponse(request, response)
+        actionHistory.trackActionResponse(response, null, workflowEngine.settings)
+        workflowEngine.recordAction(actionHistory)
```

Comment:
Add the actionHistory behavior to the `process` function instead. That will let us track the async nature of the pipelines.

Reply:
I was thinking we should keep some of the logic in `convert` in addition to having it be in `process`, since `request`/`response` are only available in `convert`.